src/h3/mod.rs

```diff
@@ -582,6 +589,43 @@ impl Connection {
         Ok(())
     }
 
+    fn send_grease_frames(
+        &mut self, conn: &mut super::Connection, stream_id: u64,
+    ) -> Result<()> {
+        let mut d = [42; 128];
+
+        let mut b = octets::Octets::with_slice(&mut d);
+        b.put_varint(0)?;
+        b.put_u8(0xb)?;
+
+        trace!(
+            "{} sending GREASE frame of type {:x} on stream {}",
+            conn.trace_id(),
+            0xb,
+            stream_id
+        );
+
+        let off = b.off();
+        conn.stream_send(stream_id, &d[..off], false)?;
+
+        let mut b = octets::Octets::with_slice(&mut d);
```

Comment:
Why do you need to recreate this? Can just encode both frames and write them together.

Reply:
But also, you don't really need to write into the hardcoded buffer (except for the varints), and can just write the single values directly to the stream (like for HEADERS and DATA).

Reply:
done